plugins/org.python.pydev.ast/src/org/python/pydev/ast/codecompletion/revisited/AbstractASTManager.java

```diff
@@ -995,6 +1015,41 @@ public TokensList getCompletionsFromTokenInLocalScope(IModule module, ICompletio
         return null;
     }
 
+    @Override
+    public boolean isNodeTypingUnionSubscript(IModule module, Object node) {
+        if (node instanceof Subscript) {
+            Subscript subscript = (Subscript) node;
+
+            String rep = NodeUtils.getFullRepresentationString(subscript);
+            if (rep == null || rep.isBlank()) {
+                return false;
+            }
+
+            TokensList importedModules = module.getTokenImportedModules();
```

Comment:
This seems a bit strange to me. Why are you checking if there's a `Union` import at all? I think you should just be checking only the actual subscript, not really any import...